massa-grpc/src/tests/stream.rs

```diff
@@ -0,0 +1,1580 @@
+        .unwrap()
+        .unwrap();
+
+    match result.result.unwrap() {
+        massa_proto_rs::massa::api::v1::send_operations_response::Result::Error(err) => {
+            assert_eq!(err.message, "too many operations per message");
+        }
+        _ => {
+            panic!("should be error");
+        }
+    }
+
+    stop_handle.stop();
+}
+
+#[tokio::test]
+#[serial]
+async fn send_endorsements() {
+    let addr: SocketAddr = "[::]:4024".parse().unwrap();
+    let mut public_server = grpc_public_service(&addr);
+    let config = public_server.grpc_config.clone();
+
+    let mut protocol_ctrl = MockProtocolController::new();
+    protocol_ctrl.expect_clone_box().returning(|| {
+        let mut ctrl = MockProtocolController::new();
+
+        ctrl.expect_propagate_endorsements().returning(|_| Ok(()));
+
+        Box::new(ctrl)
+    });
+
+    let mut pool_ctrl = MockPoolCtrl::new();
+    pool_ctrl.expect_clone_box().returning(|| {
+        let mut ctrl = MockPoolCtrl::new();
+
+        ctrl.expect_add_endorsements().returning(|_| ());
+
+        Box::new(ctrl)
+    });
+
+    public_server.pool_controller = Box::new(pool_ctrl);
+    public_server.protocol_controller = Box::new(protocol_ctrl);
+
+    let (tx, rx) = tokio::sync::mpsc::channel(10);
+    let request_stream = tokio_stream::wrappers::ReceiverStream::new(rx);
+
+    let stop_handle = public_server.serve(&config).await.unwrap();
+
+    let mut public_client = PublicServiceClient::connect(format!(
+        "grpc://localhost:{}",
+        addr.to_string().split(':').into_iter().last().unwrap()
+    ))
+    .await
+    .unwrap();
+
+    let mut resp_stream = public_client
+        .send_endorsements(request_stream)
+        .await
+        .unwrap()
+        .into_inner();
+
+    let endorsement = create_endorsement();
+    // serialize endorsement
+    let mut buffer: Vec<u8> = Vec::new();
+    SecureShareSerializer::new()
+        .serialize(&endorsement, &mut buffer)
+        .unwrap();
+
+    tx.send(SendEndorsementsRequest {
+        endorsements: vec![buffer],
+    })
+    .await
+    .unwrap();
+
+    let result = tokio::time::timeout(Duration::from_secs(5), resp_stream.next())
+        .await
+        .unwrap()
+        .unwrap()
+        .unwrap();
+
+    assert!(result.result.is_some());
+
+    // cause fail deserialize endorsement
+    tx.send(SendEndorsementsRequest {
+        endorsements: vec![endorsement.serialized_data],
+    })
+    .await
+    .unwrap();
+
+    let result = tokio::time::timeout(Duration::from_secs(5), resp_stream.next())
+        .await
+        .unwrap()
+        .unwrap()
+        .unwrap();
+
+    match result.result.unwrap() {
+        massa_proto_rs::massa::api::v1::send_endorsements_response::Result::Error(err) => {
+            assert!(err.message.contains("failed to deserialize endorsement"))
+        }
+        _ => panic!("should be error"),
+    }
+
+    stop_handle.stop();
+}
+
+#[tokio::test]
+#[serial]
+async fn send_blocks() {
```

Comment:
We should disable this one ? as we are planning to disable the `SendBlocks`

Reply:
No, I'm testing that it's unavailable